tests/std/tests/P1522R1_difference_type/test.cpp

```diff
@@ -0,0 +1,1021 @@
+    static_assert(std::numeric_limits<_Uint128>::digits == 128);
+    static_assert(std::numeric_limits<_Uint128>::is_modulo);
+
+    static_assert(std::same_as<std::common_type_t<bool, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<char, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<signed char, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<unsigned char, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<wchar_t, _Uint128>, _Uint128>);
+#ifdef __cpp_char8_t
+    static_assert(std::same_as<std::common_type_t<char8_t, _Uint128>, _Uint128>);
+#endif // __cpp_char8_t
+    static_assert(std::same_as<std::common_type_t<char16_t, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<char32_t, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<short, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<unsigned short, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<int, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<unsigned int, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<long, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<unsigned long, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<long long, _Uint128>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<unsigned long long, _Uint128>, _Uint128>);
+
+    static_assert(std::same_as<std::common_type_t<_Uint128, bool>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, char>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, signed char>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, unsigned char>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, wchar_t>, _Uint128>);
+#ifdef __cpp_char8_t
+    static_assert(std::same_as<std::common_type_t<_Uint128, char8_t>, _Uint128>);
+#endif // __cpp_char8_t
+    static_assert(std::same_as<std::common_type_t<_Uint128, char16_t>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, char32_t>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, short>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, unsigned short>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, int>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, unsigned int>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, long>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, unsigned long>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, long long>, _Uint128>);
+    static_assert(std::same_as<std::common_type_t<_Uint128, unsigned long long>, _Uint128>);

```

Comment:
What about `std::common_type_t<_Uint128, _Int128>`?

Reply:
Bug! I need to define it now that I've removed the inheritance relationship.